src/components/TooltipIcon/TooltipIcon.tsx

```diff
@@ -11,17 +11,40 @@ import { Tooltip, TooltipProps } from "@components/Tooltip/Tooltip";
 import { TooltipArrow } from "@components/Tooltip/TooltipArrow";
 import { FOCUS_STYLE } from "@utilities/focusStyle";
 import { merge } from "@utilities/merge";
+import { IconProps } from "@foundation/Icon";
 
 export type TooltipIconProps = {
     tooltip?: Omit<TooltipProps, "tooltipAriaProps">;
     iconSize?: IconSize;
+    triggerIcon?: ReactElement<IconProps>;
+    triggerStyle?: TooltipIconTriggerStyle;
 };
 
+export type TooltipIconTriggerStyle = "Danger" | "Warning" | "Primary";
+
+const getTooltipTriggerStyleClass = (style: TooltipIconTriggerStyle, isOpen: boolean) =>
```

Comment:
I'm afraid the triggerStyle part conflicts with the refactoring I'm working on. I moved the hover and open/close logic inside the tooltip component. on the new version, the TooltipIcon component no longer uses isOpen. 

I do believe you can simply use the new tooltip component and give the trigger a hover: class. And tooltipIcon can be slowly deprecated.